src/shared/libfido2-util.c

```diff
@@ -301,7 +307,12 @@ static int fido2_use_hmac_hash_specific_token(
 
                 log_info("Security token requires user presence.");
 
-                r = sym_fido_dev_get_assert(d, a, NULL); /* try without pin but with up now */
+                /* If the client pin is required, do not attempt to skip it, or it will work but give
+                 * a different secret */
+                if (FLAGS_SET(required, FIDO2ENROLL_PIN))
```

Comment:
given that we *know* that we need a PIN we should never try without one. previously we didn't store the fact that we need a PIN in the JSON data. but we do now, hence we really shouldn't bother at all with settings that don't matc what we know from the json data.